src/neural_de.jl

```diff
@@ -34,29 +45,15 @@ Ref
 [1]L. S. Pontryagin, Mathematical Theory of Optimal Processes. CRC Press, 1987.
 
 """
-struct NeuralODE{M,P,RE,T,S,A,K} <: NeuralDELayer
-    model::M
-    p::P
-    re::RE
-    tspan::T
-    solver::S
-    args::A
-    kwargs::K
-
-    function NeuralODE(model,tspan,solver=nothing,args...;kwargs...)
-        p,re = Flux.destructure(model)
-        new{typeof(model),typeof(p),typeof(re),
-            typeof(tspan),typeof(solver),typeof(args),typeof(kwargs)}(
-            model,p,re,tspan,solver,args,kwargs)
-    end
+function NeuralODE(model,tspan,solver=nothing,args...;kwargs...)
+    p,re = Flux.destructure(model)
+    NeuralODE(model,p,re,tspan,solver,args;kwargs)
```

Comment:
```suggestion
    NeuralODE(model,p,re,tspan,solver,args,kwargs)
```

but won't this infinite loop? I think the infinite looping due to the ambiguity on the splatted args is probably the reason they were made into inner constructors.

Reply:
I made that change because I was getting confused how the parser would work since both args and kwargs are splatted.

Reply:
This works:

```
function testArgs(args...; kwargs...)
  @show args
  @show kwargs
end

testArgs(1, 2, 3; a = 4, b = 5, c = 6)

args = (1, 2, 3)
kwargs = Base.Iterators.Pairs(:a => 4,:b => 5,:c => 6)
```

This fails with `syntax: invalid "..." on non-final argument`

```
function testArgs(args..., kwargs...)
  @show args
  @show kwargs
end
```

Reply:
Or you are referring to the fact that there is already a keyword before for `solver`?

Then would that signature work:
`model::FastChain, tspan, args...; solver = nothing, kwargs...`

In which case, the first 3 work, the last one throws the same error.
```
function testArgs(args; a = 5, kwargs...)
  @show args
  @show a
  @show kwargs
end
testArgs(1; a = 3, b = 5, c = 6)

function testArgs(args, a = 5, kwargs...)
  @show args
  @show a
  @show kwargs
end
testArgs(1, a = 3, b = 5, c = 6)

function testArgs(args...; a = 5, kwargs...)
  @show args
  @show a
  @show kwargs
end
testArgs(1, 2; a = 3, b = 5, c = 6)

function testArgs(args..., a = 5, kwargs...)
  @show args
  @show a
  @show kwargs
end
testArgs(1, 2; a = 3, b = 5, c = 6)
```